### ISSUE TITLE
[prefer-includes] False positive within includes implementation.

### ISSUE THREAD

**Issue by topaxi** [typescript-eslint/typescript-eslint#471]
When implementing your own data structure and `includes()` calls `indexOf()`, this rule shouldn't suggest to use `includes()`

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-includes": "error"
  }
}
```

```typescript
class MyList<T> {
  indexOf(value: T): number {
    return -1;
  }

  includes(value: T): boolean {
    return this.indexOf(value) !== -1
  }
}
```

**Expected Result**

No error

**Actual Result**

error  Use 'includes()' method instead

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.7.0` |
| `@typescript-eslint/parser`        | `1.7.0` |
| `TypeScript`                       | `3.x.x` |
| `ESLint`                           | `5.16.x` |
| `node`                             | `10.x.x` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
I'm going to close this because:
- the effort to handle this case is non-trivial
- it's an edge case that's easily handled by a disable comment
